modules/cli/src/main/scala/scala/cli/exportCmd/JsonProject.scala

```diff
@@ -0,0 +1,92 @@
+package scala.cli.exportCmd
+
+import com.github.plokhotnyuk.jsoniter_scala.core.{JsonValueCodec, WriterConfig, writeToStream}
+import com.github.plokhotnyuk.jsoniter_scala.macros.{CodecMakerConfig, JsonCodecMaker}
+import coursier.Dependency
+import coursier.util.Artifact
+
+import java.nio.charset.StandardCharsets
+
+import scala.build.options.ConfigMonoid
+import scala.cli.util.SeqHelpers.*
+import scala.reflect.NameTransformer
+import scala.util.{Properties, Using}
+
+final case class JsonProject(
+  projectName: Option[String] = None,
+  scalaVersion: Option[String] = None,
+  platform: Option[String] = None,
+  scalacOptions: Seq[String] = Nil,
+  scalaCompilerPlugins: Seq[String] = Nil,
+  scalaJsVersion: Option[String] = None,
+  scalaNativeVersion: Option[String] = None,
+  mainDeps: Seq[ExportDependencyFormat] = Nil,
+  testDeps: Seq[ExportDependencyFormat] = Nil,
+  mainSources: Seq[String] = Nil,
+  testSources: Seq[String] = Nil,
+  resolvers: Seq[ExportResolverFormat] = Nil,
+  extraDecls: Seq[String] = Nil,
+  resourcesDirs: Seq[String] = Nil,
+  mainClass: Option[String] = None
+) extends Project {
+
+  def +(other: JsonProject): JsonProject =
+    JsonProject.monoid.orElse(this, other)
+
+  def writeTo(dir: os.Path): Unit = {
+    val config = WriterConfig.withIndentionStep(1)
+
+    Using(os.write.outputStream(dir / "export.json")) {
+      outputStream =>
+        writeToStream(
+          this,
+          outputStream,
+          config
+        )
+    }
+  }
+}
+
+case class ExportResolverFormat(
+  name: String,
+  location: Option[String],
+  user: Option[String] = None,
+  password: Option[String] = None,
+  headers: Seq[(String, String)] = Nil
+)
```

Comment:
Just like for dependencies, that looks too heavyweight. A simple string should be enough to describe a repository non-ambiguously. For `MavenRepository` instances, just the URL location is enough, and will be parsed back as a `MavenRepository`. For `IvyRepository`, they have a `ivy:…` representation too, see how these are parsed [here](https://github.com/coursier/coursier/blob/b8b48dc1809726b52e0ccaa054a87211e9c98344/modules/coursier/shared/src/main/scala/coursier/internal/SharedRepositoryParser.scala#L37-L47).

About authentication, I wouldn't put it in that file by default, as this could unexpectedly leak passwords, and users using these info and fetching things with coursier down-the-line will recover the globally-configured authentication anyway, via the global coursier config files. I'd add authentication only if users explicitly ask for it.

So overall, I'd represent repositories as a simple string by default. If users ask for authentication, an object should do, with the former string in a field such as `"repo"`, and authentication fields alongside it.